src/proto/grpc/testing/control.proto

```diff
@@ -37,17 +37,32 @@ package grpc.testing;
 enum ClientType {
   SYNC_CLIENT = 0;
   ASYNC_CLIENT = 1;
+  CLIENT_VARIANT_1 = 2;
```

Comment:
@vjpai 
@jtattermusch 

Instead of having an enum could we ask the worker to return a list of names for the different types and then pass those names to initiate scenarios. In java for example we would have

server-variants: "netty-pooled", "netty-direct"
client-variants: "netty-pooled", "netty-direct", "android-okhttp"

We don't necessarily need the full M X N of these scenarios so it might be better to return a tuple of the useful combinations  perhaps annotated that with an enum indicating which driver scenarios these tuples should be applied to

E.g. 
[{ 
   server: netty-direct,
   client: netty-direct,
   scenario-type: latency
},{ 
   server: netty-pooled,
   client: netty-pooled,
   scenario-type: throughput
},

The generic vs. non-generic is orthogonal to this choice and seems like it should be rolled into RpcType